Lesson3/Task1/BankAccount.cs

```diff
@@ -0,0 +1,175 @@
+        Consolidated,
+        /// <summary>
+        /// Ссудный счёт
+        /// </summary>
+        Loan,
+        /// <summary>
+        /// Текущий счёт
+        /// </summary>
+        Current,
+        /// <summary>
+        /// Транзитный счёт
+        /// </summary>
+        Transit,
+        /// <summary>
+        /// Фидуциарный счёт
+        /// </summary>
+        Fiduciary,
+        /// <summary>
+        /// Фондовый счёт
+        /// </summary>
+        Stock,
+        /// <summary>
+        /// Частный счёт
+        /// </summary>
+        Private,
+    }
+    const int NumberStart = 1000555;
+
+    /// <summary>
+    /// Переводит запрошенную сумму со счета [source]
+    /// </summary>
+    /// <param name="source">счет источник</param>
+    /// <param name="money"> сумма перевода</param>
+    /// <returns>true - операция успешна</returns>
+    public bool Transfer(BankAccount source, decimal money)
+    {
+        if (source == null) return false;
+        if (!source.GetMoney(money)) return false;
+        _Balance += money;
+        return true;
+    }
+
+
+
+    /// <summary>
+    /// конструктор по умолчанию
+    /// </summary>
+    public BankAccount()
+    {
+        SetNumber();
+        _Balance = 0;
+        _Type = BankAccountType.Savings;
+    }
+
+    public BankAccount(decimal Balance)
+    {
+        SetNumber();
+        _Balance = Balance;
+        _Type = BankAccountType.Savings;
+    }
+
+    public BankAccount(BankAccountType type)
+    {
+        SetNumber();
+        _Balance = 0;
+        _Type = type;
+    }
+
+    public BankAccount(decimal balance, BankAccountType type)
+    {
+        SetNumber();
+        _Balance = balance;
+        _Type = type;
+    }
+
+    public int Number { get => _Number; }
+    public decimal Balance { get => _Balance; set => _Balance = value; }
+    public BankAccountType Type { get => _Type; set => _Type = value; }
+
+    /// <summary>
+    /// положить деньги на счет
+    /// </summary>
+    /// <param name="money"></param>
+    /// <returns>true - если все OK</returns>
+    public bool PutMoney(decimal money)
+    {
+        if (money < 0) return false;
+        _Balance += money;
+        return true;
+    }
+
+    /// <summary>
+    /// снять деньги со счета
+    /// </summary>
+    /// <param name="money"></param>
+    /// <returns>true - если все OK</returns>
+    public bool GetMoney(decimal money)
+    {
+        if (money < 0) return false;
+        if (money > _Balance) return false;
+        _Balance -= money;
+        return true;
+    }
+
+
+    public override string ToString()
+    {
+        return $"Номер счета: {_Number}\t\tБаланс: {_Balance}\t\tТип: {_Type}";
+    }
+
+
+    private static int _NumberUnical = NumberStart;
+    private int _Number;
+    private decimal _Balance;
+    private BankAccountType _Type;
+
+    /// <summary>
+    /// создает уникальный номер
+    /// </summary>
+    private void SetNumber() => _Number = _NumberUnical++;
```

Comment:
Плохо что метод позволяет изменить номер счёта в любой момент (когда бы его не вызвали). Метод имеет скрытые внутри него последствия вызова. Нужно переделать так, что бы метод просто возвращал уникальный номер счёта.
```suggestion
    private static int CreateNumber() =>_NumberUnical++;
```